modules/tax/src/Plugin/Commerce/TaxType/CanadianSalesTax.php

```diff
@@ -0,0 +1,261 @@
+        ['country_code' => 'CA', 'administrative_area' => 'NL']
+      ],
+      'rates' => [
+        [
+          'id' => 'HST',
+          'label' => $this->t('HST'),
+          'amounts' => [
+            ['amount' => '0.15', 'start_date' => '2016-07-01'],
+          ]
+        ]
+      ]
+    ]);
+
+    $zones['ns'] = new TaxZone([
+      'id' => 'ns',
+      'label' => $this->t('Nova Scotia Harmonized Sales Tax'),
+      'territories' => [
+        ['country_code' => 'CA', 'administrative_area' => 'NS']
+      ],
+      'rates' => [
+        [
+          'id' => 'HST',
+          'label' => $this->t('HST'),
+          'amounts' => [
+            ['amount' => '0.15', 'start_date' => '2010-07-01'],
+          ]
+        ]
+      ]
+    ]);
+
+    $zones['on'] = new TaxZone([
+      'id' => 'on',
+      'label' => $this->t('Ontario Harmonized Sales Tax'),
+      'territories' => [
+        ['country_code' => 'CA', 'administrative_area' => 'ON']
+      ],
+      'rates' => [
+        [
+          'id' => 'HST',
+          'label' => $this->t('HST'),
+          'amounts' => [
+            ['amount' => '0.13', 'start_date' => '2010-07-01'],
+          ]
+        ]
+      ]
+    ]);
+
+    $zones['pe'] = new TaxZone([
+      'id' => 'pe',
+      'label' => $this->t('PEI Harmonized Sales Tax'),
+      'territories' => [
+        ['country_code' => 'CA', 'administrative_area' => 'PE']
+      ],
+      'rates' => [
+        [
+          'id' => 'HST',
+          'label' => $this->t('HST'),
+          'amounts' => [
+            ['amount' => '0.14', 'start_date' => '2013-04-01'],
+          ]
+        ]
+      ]
+    ]);
+
+    $zones['qc'] = new TaxZone([
+      'id' => 'qc',
+      'label' => $this->t('Quebec Sales Tax'),
+      'territories' => [
+        ['country_code' => 'CA', 'administrative_area' => 'QC']
+      ],
+      'rates' => [
+        [
+          'id' => 'QST',
+          'label' => $this->t('QST'),
+          'amounts' => [
+            ['amount' => '0.09975', 'start_date' => '2013-01-01'],
+          ]
+        ]
+      ]
+    ]);
+
+    return $zones;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function resolveZones(OrderItemInterface $order_item, ProfileInterface $customer_profile) {
+    $zones = $this->getZones();
+    $customer_address = $customer_profile->address->first();
+    $customer_zones = array_filter($zones, function ($zone) use ($customer_address) {
+      /** @var \Drupal\commerce_tax\TaxZone $zone */
+      return $zone->match($customer_address);
+    });
+    if (empty($customer_zones)) {
+      // The customer is not in Canada.
+      return [];
+    }
+    $order = $order_item->getOrder();
+    $store = $order->getStore();
+    $store_address = $store->getAddress();
+    $store_zones = array_filter($zones, function ($zone) use ($store_address) {
+      /** @var \Drupal\commerce_tax\TaxZone $zone */
+      return $zone->match($store_address);
+    });
+
+    $resolved_zones = $store_zones;
```

Comment:
This doesn't match the tax library logic, you want the customer zone, not the store zone.